### PR TITLE
[T2] removed duplicated schemas in storage

### PR DESCRIPTION
**Pull request by zikalino**
Duplicated schemas are causing errors in T2 generators

### CODE REVIEW

File: specification/storage/resource-manager/Microsoft.Storage/stable/2016-12-01/storage.json [hunk 1]
```diff
@@ -1191,6 +1191,10 @@
             "f",
             "s"
           ],
+          "x-ms-enum": {
```
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2016-12-01/storage.json:1194]: why do this change?
It seems not related with the PR title.

**Review by blueww — APPROVED** [review]

**Review by HimanshuChhabra — APPROVED** [review]

File: specification/storage/resource-manager/Microsoft.Storage/stable/2016-12-01/storage.json (hunk 1 above)
Comment (zikalino) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2016-12-01/storage.json:1194]: it is related. Seems that  "x-ms-client-name": "Resource" is causing enum to be renamed to Resource.

There's also Resource defined here:

https://github.com/Azure/azure-rest-api-specs/blob/f9d1625a69e739f362dd16e32f8f9e8fa01abd37/specification/storage/resource-manager/Microsoft.Storage/stable/2016-12-01/storage.json#L1057

Here I have aligned this definition with version 2017-06-01:

https://github.com/Azure/azure-rest-api-specs/blob/f9d1625a69e739f362dd16e32f8f9e8fa01abd37/specification/storage/resource-manager/Microsoft.Storage/stable/2017-06-01/storage.json#L1771

Anyway, it's a bit awkward, if I am correct:
- original name is "signedResource"
- then it's renamed to "resource" by: "x-ms-client-name": "Resource"
- and then overriden back by:  "name": "signedResource"

I have to check if it's causing any breaking changes in 2016-12-01 and if it can be simplified in following versions.

I am not sure why it was working with previous versions of autorest, but now seems to be detected and causing SDK generation problems.

Comment (zikalino) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2016-12-01/storage.json:1194]: i have checked, and the change doesn't introduce breaking change, before definition of params was:

    :type resource: str or ~azure.mgmt.storage.v2016_12_01.models.enum

now it becomes:

    :type resource: str or
     ~azure.mgmt.storage.v2016_12_01.models.SignedResource

Simply new type is introduced
